dali/test/python/test_pipeline_debug.py

```diff
@@ -533,3 +516,70 @@ def test_multiple_input_sets():
     pipe_standard = multiple_input_sets_pipeline()
     pipe_debug = multiple_input_sets_pipeline(debug=True)
     compare_pipelines(pipe_standard, pipe_debug, 8, 10)
+
+
+@pipeline_def(batch_size=8, num_threads=3, device_id=0, seed=47, debug=True)
+def variable_batch_size_from_external_source_pipeline(variable_batch_size):
```

Comment:
This just tests one iteration to be smaller than the max_batch_size, but not exactly changing it iter2iter, right?

Reply:
Yes, added more iterations with changing batch size